### PR TITLE
Backport multiple dependency updates to branch-5.0

### PR DESCRIPTION
**Pull request by avelanarius**
Backport multiple commits that update dependencies:
- logback
- jackson (multiple packages of this family)
- snakeyaml

Additionally, netty dependency is updated (not a part of any upstream Cassandra commit per se).

### CODE REVIEW

**Comment by avelanarius** [comment]
Output of `trivy` scanner (`trivy -d rootfs lib`) before this PR:

```
Java (jar)
==========
Total: 12 (UNKNOWN: 0, LOW: 1, MEDIUM: 4, HIGH: 5, CRITICAL: 2)

+-----------------------------------------+------------------+----------+-------------------+----------------------------+--------------------------------------------+
|                 LIBRARY                 | VULNERABILITY ID | SEVERITY | INSTALLED VERSION |       FIXED VERSION        |                   TITLE                    |
+-----------------------------------------+------------------+----------+-------------------+----------------------------+--------------------------------------------+
| ch.qos.logback:logback-classic          | CVE-2017-5929    | CRITICAL | 1.1.3             | 1.2.0                      | logback: Serialization                     |
| (logback-classic-1.1.3.jar)             |                  |          |                   |                            | vulnerability in SocketServer              |
|                                         |                  |          |                   |                            | and ServerSocketReceiver                   |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2017-5929       |
+-----------------------------------------+                  +          +                   +                            +                                            +
| ch.qos.logback:logback-core             |                  |          |                   |                            |                                            |
| (logback-core-1.1.3.jar)                |                  |          |                   |                            |                                            |
|                                         |                  |          |                   |                            |                                            |
|                                         |                  |          |                   |                            |                                            |
+                                         +------------------+----------+                   +----------------------------+--------------------------------------------+
|                                         | CVE-2021-42550   | MEDIUM   |                   | 1.2.9                      | logback: remote code execution             |
|                                         |                  |          |                   |                            | through JNDI call from within              |
|                                         |                  |          |                   |                            | its configuration file...                  |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2021-42550      |
+-----------------------------------------+------------------+          +-------------------+----------------------------+--------------------------------------------+
| com.google.guava:guava                  | CVE-2018-10237   |          |              18.0 | 24.1.1-jre, 24.1.1-android | guava: Unbounded memory                    |
| (guava-18.0.jar)                        |                  |          |                   |                            | allocation in AtomicDoubleArray            |
|                                         |                  |          |                   |                            | and CompoundOrdering classes               |
|                                         |                  |          |                   |                            | allow remote attackers...                  |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2018-10237      |
+                                         +------------------+----------+                   +----------------------------+--------------------------------------------+
|                                         | CVE-2020-8908    | LOW      |                   |                       30.0 | guava: local information                   |
|                                         |                  |          |                   |                            | disclosure via temporary directory         |
|                                         |                  |          |                   |                            | created with unsafe permissions            |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2020-8908       |
+-----------------------------------------+------------------+----------+-------------------+----------------------------+--------------------------------------------+
| io.netty:netty-all                      | CVE-2019-16869   | HIGH     | 4.0.56.Final      | 4.1.42.Final               | netty: HTTP request smuggling              |
| (netty-all-4.0.56.Final.jar)            |                  |          |                   |                            | by mishandled whitespace                   |
|                                         |                  |          |                   |                            | before the colon in HTTP...                |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2019-16869      |
+-----------------------------------------+------------------+          +-------------------+----------------------------+--------------------------------------------+
| org.apache.thrift:libthrift             | CVE-2018-1320    |          | 0.9.2             | 0.12.0                     | thrift: SASL negotiation                   |
| (libthrift-0.9.2.jar)                   |                  |          |                   |                            | isComplete validation bypass in the        |
|                                         |                  |          |                   |                            | org.apache.thrift.transport.TSaslTransport |
|                                         |                  |          |                   |                            | class -->avd.aquasec.com/nvd/cve-2018-1320 |
+                                         +------------------+          +                   +----------------------------+--------------------------------------------+
|                                         | CVE-2019-0205    |          |                   | 0.13.0                     | thrift: Endless loop when                  |
|                                         |                  |          |                   |                            | feed with specific input data              |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2019-0205       |
+                                         +------------------+----------+                   +----------------------------+--------------------------------------------+
|                                         | CVE-2015-3254    | MEDIUM   |                   | 0.9.3                      | thrift: Infinite recursion via             |
|                                         |                  |          |                   |                            | vectors involving the skip function        |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2015-3254       |
+                                         +------------------+          +                   +----------------------------+--------------------------------------------+
|                                         | CVE-2018-11798   |          |                   | 0.12.0                     | thrift: Improper Access                    |
|                                         |                  |          |                   |                            | Control grants access to files             |
|                                         |                  |          |                   |                            | outside the  webservers...                 |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2018-11798      |
+-----------------------------------------+------------------+----------+-------------------+----------------------------+--------------------------------------------+
| org.codehaus.jackson:jackson-mapper-asl | CVE-2019-10172   | HIGH     | 1.9.13            |                            | jackson-mapper-asl: XML external           |
| (jackson-mapper-asl-1.9.13.jar)         |                  |          |                   |                            | entity similar to CVE-2016-3720            |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2019-10172      |
+-----------------------------------------+------------------+          +-------------------+----------------------------+--------------------------------------------+
| org.yaml:snakeyaml                      | CVE-2017-18640   |          |              1.11 |                       1.26 | snakeyaml: Billion laughs                  |
| (snakeyaml-1.11.jar)                    |                  |          |                   |                            | attack via alias feature                   |
|                                         |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2017-18640      |
+-----------------------------------------+------------------+----------+-------------------+----------------------------+--------------------------------------------+
```

after this PR:
```
Java (jar)
==========
Total: 6 (UNKNOWN: 0, LOW: 1, MEDIUM: 3, HIGH: 2, CRITICAL: 0)

+--------------------------------+------------------+----------+-------------------+----------------------------+--------------------------------------------+
|            LIBRARY             | VULNERABILITY ID | SEVERITY | INSTALLED VERSION |       FIXED VERSION        |                   TITLE                    |
+--------------------------------+------------------+----------+-------------------+----------------------------+--------------------------------------------+
| com.google.guava:guava         | CVE-2018-10237   | MEDIUM   |              18.0 | 24.1.1-jre, 24.1.1-android | guava: Unbounded memory                    |
| (guava-18.0.jar)               |                  |          |                   |                            | allocation in AtomicDoubleArray            |
|                                |                  |          |                   |                            | and CompoundOrdering classes               |
|                                |                  |          |                   |                            | allow remote attackers...                  |
|                                |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2018-10237      |
+                                +------------------+----------+                   +----------------------------+--------------------------------------------+
|                                | CVE-2020-8908    | LOW      |                   |                       30.0 | guava: local information                   |
|                                |                  |          |                   |                            | disclosure via temporary directory         |
|                                |                  |          |                   |                            | created with unsafe permissions            |
|                                |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2020-8908       |
+--------------------------------+------------------+----------+-------------------+----------------------------+--------------------------------------------+
| org.apache.thrift:libthrift    | CVE-2018-1320    | HIGH     | 0.9.2             | 0.12.0                     | thrift: SASL negotiation                   |
| (libthrift-0.9.2.jar)          |                  |          |                   |                            | isComplete validation bypass in the        |
|                                |                  |          |                   |                            | org.apache.thrift.transport.TSaslTransport |
|                                |                  |          |                   |                            | class -->avd.aquasec.com/nvd/cve-2018-1320 |
+                                +------------------+          +                   +----------------------------+--------------------------------------------+
|                                | CVE-2019-0205    |          |                   | 0.13.0                     | thrift: Endless loop when                  |
|                                |                  |          |                   |                            | feed with specific input data              |
|                                |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2019-0205       |
+                                +------------------+----------+                   +----------------------------+--------------------------------------------+
|                                | CVE-2015-3254    | MEDIUM   |                   | 0.9.3                      | thrift: Infinite recursion via             |
|                                |                  |          |                   |                            | vectors involving the skip function        |
|                                |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2015-3254       |
+                                +------------------+          +                   +----------------------------+--------------------------------------------+
|                                | CVE-2018-11798   |          |                   | 0.12.0                     | thrift: Improper Access                    |
|                                |                  |          |                   |                            | Control grants access to files             |
|                                |                  |          |                   |                            | outside the  webservers...                 |
|                                |                  |          |                   |                            | -->avd.aquasec.com/nvd/cve-2018-11798      |
+--------------------------------+------------------+----------+-------------------+----------------------------+--------------------------------------------+
```

Please refer to https://github.com/scylladb/scylla-tools-java/pull/297#issuecomment-1136409898 for the reasons why `guava` and `libthrift` are not updated.

**Comment by avelanarius** [comment]
This PR applies cleanly also to `branch-4.3` and produces the same `trivy` report as on `branch-5.0`.

**Review by elcallio — APPROVED** [review]
lgtm

**Comment by slivne** [comment]
@scylladb/scylla-maint ping

File: build.xml [hunk 1]
```diff
@@ -486,7 +486,7 @@
           <dependency groupId="com.addthis.metrics" artifactId="reporter-config3" version="3.0.3" />
           <dependency groupId="org.mindrot" artifactId="jbcrypt" version="0.3m" />
           <dependency groupId="io.airlift" artifactId="airline" version="0.6" />
-          <dependency groupId="io.netty" artifactId="netty-all" version="4.0.56.Final" />
+          <dependency groupId="io.netty" artifactId="netty-all" version="4.1.42.Final" />
```
Comment (avikivity) [inline build.xml:490]: Please update the patch changelog with paperwork (cherry-picked-from or something else) that explains why master is okay. We have a trickle-down policy where everything must be fixed in master first, so an upgrade never creates a regression.

File: build.xml [hunk 2]
```diff
@@ -436,9 +436,9 @@
           <dependency groupId="org.slf4j" artifactId="jcl-over-slf4j" version="1.7.7" />
           <dependency groupId="ch.qos.logback" artifactId="logback-core" version="1.2.9"/>
           <dependency groupId="ch.qos.logback" artifactId="logback-classic" version="1.2.9"/>
-          <dependency groupId="com.fasterxml.jackson.core" artifactId="jackson-core" version="2.9.10"/>
-          <dependency groupId="com.fasterxml.jackson.core" artifactId="jackson-databind" version="2.9.10.4"/>
-          <dependency groupId="com.fasterxml.jackson.core" artifactId="jackson-annotations" version="2.9.10"/>
+          <dependency groupId="com.fasterxml.jackson.core" artifactId="jackson-core" version="2.13.2"/>
+          <dependency groupId="com.fasterxml.jackson.core" artifactId="jackson-databind" version="2.13.2.2"/>
```
Comment (avikivity) [inline build.xml:440]: Again need reference to our master.

Comment (avelanarius) [reply build.xml:440]: Once https://github.com/scylladb/scylla-tools-java/pull/297 is merged, only then I will be able to reference master.

**Comment by mykaul** [comment]
@avelanarius - what's the status here?

**Comment by mykaul** [comment]
@avelanarius ?

File: build.xml (hunk 2 above)
Comment (mykaul) [reply build.xml:440]: @avelanarius - the above was merged long time ago.

**Comment by mykaul** [comment]
@yaronkaikov - unsure if this is still needed?

**Comment by yaronkaikov** [comment]
@mykaul Not relevant , closing

**Pull request closed without merging**